packages/eas-build-job/src/steps.ts

```diff
@@ -0,0 +1,135 @@
+import { z } from 'zod';
+
+const CommonStepZ = z.object({
+  /**
+   * Unique identifier for the step.
+   *
+   * @example
+   * id: step1
+   */
+  id: z.string().optional(),
+  /**
+   * Expression that determines whether the step should run.
+   * Based on the GitHub Actions job step `if` field (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif).
+   *
+   * @example
+   * if: ${{ steps.step1.outputs.test == 'test' }}
```

Comment:
I'm not sure this is how `if`s are going to look like. https://exponent-internal.slack.com/archives/C04GPRQCEBG/p1711554721112529?thread_ts=1711392256.213759&cid=C04GPRQCEBG

I would skip adding examples if we aren't sure what things are going to look like.